allennlp/data/fields/text_field.py

```diff
@@ -66,33 +67,37 @@ def get_padding_lengths(self) -> Dict[str, int]:
             for key in token_lengths[0].keys():
                 indexer_lengths[key] = max(x[key] if key in x else 0 for x in token_lengths)
             lengths.append(indexer_lengths)
-        padding_lengths = {'num_tokens': len(self._indexed_tokens[0])}
+        any_indexed_token_key = list(self._indexed_tokens.keys())[0]
+        padding_lengths = {'num_tokens': len(self._indexed_tokens[any_indexed_token_key])}
         # Get all the keys which have been used for padding.
         padding_keys = {key for d in lengths for key in d.keys()}
         for padding_key in padding_keys:
             padding_lengths[padding_key] = max(x[padding_key] if padding_key in x else 0 for x in lengths)
         return padding_lengths
 
-    @overrides
+    # @overrides
     def sequence_length(self) -> int:
         return len(self._tokens)
 
-    @overrides
-    def pad(self, padding_lengths: Dict[str, int]) -> List[numpy.array]:
-        arrays = []
+    # @overrides
+    def as_array(self, padding_lengths: Dict[str, int]) -> Dict[str, numpy.array]:
+        arrays = {}
         desired_num_tokens = padding_lengths['num_tokens']
-        for indexer, array in zip(self._token_indexers, self._indexed_tokens):
-            padded_array = indexer.pad_token_sequence(array, desired_num_tokens, padding_lengths)
-            arrays.append(numpy.asarray(padded_array))
+        for (indexer_name, indexer) in self._token_indexers.items():
+            padded_array = indexer.pad_token_sequence(self._indexed_tokens[indexer_name],
+                                                      desired_num_tokens, padding_lengths)
+            # Use the namespace of the indexer as a key to recognise what
```

Comment:
I'd avoid the word "namespace" here - maybe just say "We use the same keys in the array dictionary as were used for the token indexers given in the constructor", or something similar.